packages/platform-core/API.md

```diff
@@ -7,6 +7,64 @@
 import { BackendIdentifier } from '@aws-amplify/plugin-types';
 import z from 'zod';
 
+// @public
+export class AmplifyError extends Error {
+    constructor(name: AmplifyErrorType, classification: AmplifyErrorClassification, options: AmplifyErrorOptions, downstreamError?: Error | undefined);
+    // (undocumented)
+    readonly classification: AmplifyErrorClassification;
+    // (undocumented)
+    readonly code?: string;
+    // (undocumented)
+    readonly details?: string;
+    // (undocumented)
+    readonly downstreamError?: Error | undefined;
+    // (undocumented)
+    static fromError: (error: unknown) => AmplifyError;
+    // (undocumented)
+    static fromStderr: (_stderr: string) => AmplifyError | undefined;
+    // (undocumented)
+    readonly link?: string;
+    // (undocumented)
+    readonly message: string;
+    // (undocumented)
+    readonly name: AmplifyErrorType;
+    // (undocumented)
+    readonly resolution?: string;
+    // (undocumented)
+    serializedError?: string;
+}
+
+// @public
+export type AmplifyErrorClassification = 'FAULT' | 'ERROR';
+
+// @public
+export type AmplifyErrorOptions = {
+    message: string;
+    details?: string;
+    resolution?: string;
+    link?: string;
+    code?: string;
+};
+
+// @public
+export type AmplifyErrorType = AmplifyUserErrorType | AmplifyLibraryFaultType;
+
+// @public
+export class AmplifyFault extends AmplifyError {
+    constructor(name: AmplifyLibraryFaultType, options: AmplifyErrorOptions, downstreamError?: Error);
+}
+
+// @public
+export type AmplifyLibraryFaultType = 'UnknownFault';
+
+// @public
+export class AmplifyUserError extends AmplifyError {
+    constructor(name: AmplifyUserErrorType, options: AmplifyErrorOptions, downstreamError?: Error);
+}

```

Comment:
We should either have `AmplifyErrorClassification` (+/- other props on `AmplifyError`) or subtypes of `AmplifyError`.
Having both is kind of redundant.

If we have subtypes - should `AmplifyError` be abstract ?

Reply:
Made `AmplifyError` abstract